71_RayTracingPipeline/app_resources/raytrace_procedural.rchit.hlsl

```diff
@@ -0,0 +1,61 @@
+#include "common.hlsl"
+
+[[vk::push_constant]] SPushConstants pc;
+
+[[vk::binding(0, 0)]] RaytracingAccelerationStructure topLevelAS;
+
+[shader("closesthit")]
+void main(inout ColorPayload p, in BuiltInTriangleIntersectionAttributes attribs)
+{
+    const int instID = InstanceID();
+    const int primID = PrimitiveIndex();
+    float32_t3 worldPosition = WorldRayOrigin() + WorldRayDirection() * RayTCurrent();
+
+    SProceduralGeomInfo sphere = vk::RawBufferLoad < SProceduralGeomInfo > (pc.proceduralGeomInfoBuffer + primID * sizeof(SProceduralGeomInfo));
+
+    // Computing the normal at hit position
+    float32_t3 worldNormal = normalize(worldPosition - sphere.center);
```

Comment:
you already had to load and compute some of this in the intersection shader, is it not possible to pass it in the payload/custom attribs to the closest hit shader?